x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicators_flyout_table/indicators_flyout_table.tsx

```diff
@@ -46,16 +48,29 @@ const search = {
 
 export interface IndicatorsFlyoutTableProps {
   indicator: Indicator;
+  fieldTypesMap: { [id: string]: string };
 }
 
-export const IndicatorsFlyoutTable: VFC<IndicatorsFlyoutTableProps> = ({ indicator }) => {
-  const items: Array<{ field: string; value: string }> = [];
+export const IndicatorsFlyoutTable: VFC<IndicatorsFlyoutTableProps> = ({
+  indicator,
+  fieldTypesMap,
+}) => {
+  const userTimeZone = useTimeZone();
+  const userFormat = useDateFormat();
 
+  const items: Array<{ field: string; value: string }> = [];
   for (const key in indicator.fields) {
     if (!indicator.fields.hasOwnProperty(key)) continue;
+
+    const fieldType: string = fieldTypesMap[key];
+    const value = unwrapValue(indicator, key as RawIndicatorFieldId);
+
     items.push({
       field: key,
-      value: unwrapValue(indicator, key as RawIndicatorFieldId) || EMPTY_VALUE,
+      value:
+        fieldType === 'date'
+          ? dateFormatter(value as string, userTimeZone, userFormat)
```

Comment:
one thing that might make the component a bit cleaner is to move this logic into value column renderer (there is a `render` field available when creating an array of columns) and there we can some of this logic and what's more important we can use <DateFormatter> component without the need to get timezone and dateformat from hooks inside the flyout table

Reply:
good idea I'll try that!

Reply:
I meant `DateFormatter` component, just missed it in the comment, just in case :)

Reply:
if I use the render field then I should be able to use the DateFormatter, indeed! I couldn't with the current implementation

Reply:
@maxcold can you take a look at [my last commit](https://github.com/elastic/kibana/pull/138106/commits/b1b29d5f7f3cbee0b1af95e3d3b9dd0e5dc40d05)? Is this what you had in mind?
We're using a table (not a grid) and it requires the item props as an array. So I need to pass some sort of array to it, and from the indicator object, I have an object of feeds.

For some reason I don't find this approach super efficient or clean. What am I doing wrong?

Reply:
actually @maxcold you can take a look at [this newest commit](https://github.com/elastic/kibana/pull/138106/commits/b596e9051d3efa31787f5caf42128346de2320b3), it seems a lot cleaner to me and also takes care of your other comment below, as I can reuse my factory in both tables. Let me know what you think of it, and if you like it I'll write some tests for it!